transmart-copy/src/main/groovy/org/transmartproject/copy/Copy.groovy

```diff
@@ -140,61 +127,90 @@ class Copy {
         relations.load(rootPath)
     }
 
-    void deleteStudy(String studyId) {
+    void deleteStudyById(String studyId, boolean failOnNoStudy = true) {
         log.info "Deleting study ${studyId} ..."
         def dimensions = new Dimensions(database)
         studies = new Studies(database, dimensions)
-        studies.delete(studyId)
+        studies.deleteById(studyId, failOnNoStudy)
         log.info "Study ${studyId} deleted."
     }
 
+    void deleteStudy(String rootPath, boolean failOnNoStudy = true) {
+        log.info "Deleting studies found in ${rootPath} ..."
+        def dimensions = new Dimensions(database)
+        studies = new Studies(database, dimensions)
+        studies.delete(rootPath, failOnNoStudy)
+        log.info "Study found in ${rootPath} deleted."
+    }
+
     static void main(String[] args) {
         def parser = new DefaultParser()
         try {
             CommandLine cl = parser.parse(options, args)
-            if (cl.hasOption('help')) {
+            if (cl.hasOption('help') || !cl.options) {
                 printHelp()
                 return
             }
-            def copy = new Copy()
+            runCopy(cl, [:].withDefault { Object key -> System.getenv((String) key) })
+        } catch (ParseException e) {
+            log.error e.message
+            println()
+            printHelp()
+            System.exit(2)
+        } catch (Exception e) {
+            log.error e.message, e
+            System.exit(1)
+        }
+    }
+
+    public final static Set<String> INDEPENDENT_OPERATION_OPTIONS = ['help', 'delete', 'drop-indexes',
+                                                                     'restore-indexes', 'vacuum-analyze'] as Set
+
+    static void runCopy(CommandLine cl, Map<String, String> params) {
+        def copy = new Copy(params)
+        def tx = copy.database.beginTransaction()
+        try {
             if (cl.hasOption('delete')) {
                 def studyId = cl.getOptionValue('delete')
-                try {
-                    copy.deleteStudy(studyId)
-                } catch(Exception e) {
-                    log.error "Error deleting study ${studyId}."
-                    throw e
-                }
+                copy.deleteStudyById(studyId)
+            }
+            if (cl.hasOption('re-upload')) {
+                def directory = cl.getOptionValue('re-upload')
+                copy.deleteStudy(directory, false)
             }
             if (cl.hasOption('drop-indexes')) {
                 copy.dropIndexes()
             }
-            if (cl.hasOption('mode')) {
+            boolean independentOperationOptionSpecified = INDEPENDENT_OPERATION_OPTIONS
```

Comment:
The control flow is really unclear here.

Reply:
It became a little bit simpler by removing `--re-upload` option ;)